### DISCUSSION TITLE
Ignore error from CLIENT SETNAME?

### DISCUSSION THREAD

**Discussion by ash2k** [redis/go-redis#2537] [General]
Hi! In https://github.com/redis/go-redis/issues/2312 a feature has been implemented to set client name via a struct field. This is cool, thanks. I tried to use it but it turned out that [GCP's Memorystore does not support it](https://issuetracker.google.com/issues/235019443), which is a shame, but what can we do. I get the following error from GCP:

```
redis: ERR unknown command `client`, with args beginning with: `setname`, `myclientname`
```

I ship my application to clients and I don't know what they use. Some might use GCP (like we ourselves do in some environments). So I cannot unconditionally use this feature because it would break the program for some customers who use GCP.

What can we do?

One idea I have is to change the library to log the error from that command instead of erroring out. Maybe check for "unknown command" text and ignore that specifically. WDYT?

**Comment by monkey92t** [comment]
Thank you for the feedback, but `ClientName` in redis.Options is an option that invokes the "client setname" command.

If you are using a version of Redis or a third-party storage that does not support this command, you can ignore this option(ClientName="").

**Reply by ash2k** [reply]
Thanks for the reply. I know I can skip setting this field, but I do want to set it so that admins see what is connecting. But I cannot just always set it because of SOME redis implementations not supporting it. As I said, I publish my program and users run it in unknown (to me) environments, some use GCP where it does not work.

I guess I could set it, try to create a client and make a test request, inspect the error and re-create the client if it's the `unknown command client, with args beginning with: setname` error... This is quite a bit of hassle. That's why I suggest the library handles this case, not me. Library could just log this error, not propagate it to the caller.

**Reply by monkey92t** [reply]
I understand what you mean, but doing it this way feels very strange.
For example, I did not set a password for the redis-server, but if you use the `AUTH password` command for authentication, the redis-server still responds with an error.

(error) ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct?

**Reply by monkey92t** [reply]
When the user sets the "clientName" option and the redis-server responds with an error, it seems even more frightening if we ignore this error and let the user mistakenly think that they have successfully executed the "CLIENT SETNAME" command.

**Reply by ash2k** [reply]
I'm suggesting to ignore (log) only a specific error - `unknown command client, with args beginning with: setname`, everything else should be propagated to the caller, as it is today.

If we're not going to do what I'm proposing, what are you suggesting I should do in my case?

**Reply by monkey92t** [reply]
I think a hook can be added. Before or after authentication, some users want to perform authentication work by themselves instead of executing go-redis authentication process (HELLO/ACL and other commands).

it's just an idea, no work has been done yet.

**Comment by taltal78** [comment]
Hi,
any update on the topic? i just bumped into into it myself on GCP...
any plans to mitigate this?
thanks,
Tal

**Reply by monkey92t** [reply]
Hi, as mentioned in the above comment, it is optional. You can ignore `ClientName`, and the client will not send the `setname` command.

**Comment by ndyakov** [comment]
You can also use `DisableIdentity` https://github.com/redis/go-redis?tab=readme-ov-file#disabling-identity-verification
